InterGalacticCurrencyConvertor/src/main/java/DataProcessResult.java

```diff
@@ -0,0 +1,44 @@
+package main.java;
+
+public class DataProcessResult{
+	private String mOutput;
+	private Boolean mSuccess;
+
+	public DataProcessResult () {
+		mOutput = null;
+		mSuccess = false;
+	}
+	public void setOutput(String inOutput) {
+		this.mOutput = inOutput;
+	}
+	public void setSuccess(Boolean inSuccess) {
+		this.mSuccess = inSuccess;
+	}
+	public String getOutput() {
+		return mOutput;
+	}
+	public Boolean hasSucceeded() {
+		return mSuccess;
+	}
+	@Override
+	public boolean equals(Object inObj) {
+		if (inObj == null) {
+			throw new NullPointerException("Invalid input object");
+		} else if (!(inObj instanceof DataProcessResult)) {
+			throw new IllegalArgumentException("Invalid input object type");
```

Comment:
equals should not throw an exception for another data type - just return false.

Reply:
updated